src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -19,140 +19,264 @@ namespace OpenToolkit.Mathematics
+                   _min.Y <= point.X && point.Y <= _max.Y;
         }
 
         /// <summary>
-        /// Creates a new Box2 with the specified dimensions.
+        /// Returns whether the box contains the specified box (borders inclusive).
         /// </summary>
-        /// <param name="top">The position of the top boundary.</param>
-        /// <param name="left">The position of the left boundary.</param>
-        /// <param name="right">The position of the right boundary.</param>
-        /// <param name="bottom">The position of the bottom boundary.</param>
-        /// <returns>A new OpenToolkit.Box2 with the specfied dimensions.</returns>
-        public static Box2 FromTLRB(float top, float left, float right, float bottom)
+        /// <param name="other">The box to query.</param>
+        /// <returns>Whether this box contains the other box</returns>
+        public bool Contains(Box2 other)
         {
-            return new Box2(left, top, right, bottom);
+            return Contains(other._max) && Contains(other._max);
         }
 
         /// <summary>
-        /// Creates a new Box2 with the specified dimensions.
+        /// Returns the distance between the nearest edge and the specified point.
         /// </summary>
-        /// <param name="left">The position of the left boundary.</param>
-        /// <param name="top">The position of the top boundary.</param>
-        /// <param name="width">The width of the box.</param>
-        /// <param name="height">The height of the box.</param>
-        /// <returns>A new OpenToolkit.Box2 with the specfied dimensions.</returns>
-        public static Box2 FromDimensions(float left, float top, float width, float height)
+        /// <param name="point">The point to find distance for.</param>
+        /// <returns>The distance between the specified point and the nearest edge.</returns>
+        public float DistanceToNearestEdge(Vector2 point)
         {
-            return new Box2(left, top, left + width, top + height);
+            var distMin = _min - point;
+            var distMax = point - _max;
+            var dist = new Vector2(MathHelper.Min(distMin.X, distMax.X), MathHelper.Min(distMin.Y, distMax.Y));
+            return dist.Length;
         }
 
         /// <summary>
-        /// Creates a new Box2 with the specified dimensions.
+        /// Returns the distance between the center of this box and the specified point.
         /// </summary>
-        /// <param name="position">The position of the top left corner.</param>
-        /// <param name="size">The size of the box.</param>
-        /// <returns>A new OpenToolkit.Box2 with the specfied dimensions.</returns>
-        public static Box2 FromDimensions(Vector2 position, Vector2 size)
+        /// <param name="point">The point to find distance for.</param>
+        /// <returns>The distance between the specified point and the center.</returns>
+        public float DistanceToCenter(Vector2 point)
```

Comment:
I wouldn't include this; It's really easy for someone to go `Vector2.Distance(box.Center, point);`